src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_copy_b.cpp

```diff
@@ -97,33 +66,25 @@ void BrgemmCopyB::custom_constructor_validate_and_infer_types(const std::vector<
 
 void BrgemmCopyB::validate_and_infer_types() {
     INTERNAL_OP_SCOPE(BrgemmRepack_validate_and_infer_types);
-    const auto& element_type = get_input_element_type(0);
-    validate_element_type(element_type);
     const auto port = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(input(0));
     const auto shape = ov::Shape(port->get_shape());
     const auto& planar_pshape = snippets::utils::get_planar_pshape(shape, port->get_layout());
```

Comment:
Should we check that `get_input_element_type(0) == m_config.orig_wei_dt()` here?

Reply:
Good idea, the check is added to both `validate_and_infer_types` and `custom_constructor_validate_and_infer_types`. Thanks!